app/models/gather_repository_maintenance_stats.rb

```diff
@@ -1,12 +1,33 @@
 class GatherRepositoryMaintenanceStats
+    def self.gather_bitbucket_stats(repository)
+        metrics = []
+
+        # get latest issues and pull requests and store them in the database
+        repository.download_issues
+        repository.download_pull_requests
+
+        metrics << MaintenanceStats::Stats::Bitbucket::CommitsStat.new(repository.retrieve_commits).get_stats
+
+        metrics << MaintenanceStats::Stats::Bitbucket::IssueRates.new(repository.issues).get_stats
+
+        add_metrics_to_repo(repository, metrics)
+        metrics
+    end
+
     def self.gather_stats(repository)
         # only support Github repos for now
         # check to make sure the Project URLs are also pointing to a Github repository
         unless stats_enabled?(repository)
```

Comment:
I think this might be easier to read as something like:
```ruby
if github?(repository)
  gather_github_stats(repository)
elsif bitbucket?(repository)
  gather_bitbucket_stats(repository)
else
  repository.repository_maintenance_stats.destroy_all
  nil
end
```